disperser/dataapi/metrics.go

```diff
@@ -236,3 +245,157 @@ func (collector *DynamoDBCollector) Collect(ch chan<- prometheus.Metric) {
+	ctx, cancel := context.WithCancel(context.Background())
+	collector := &DynamoDBCollectorV2{
+		blobMetadataStore: blobMetadataStore,
+		statusMetrics:     make(map[commonv2.BlobStatus]*BlobStatusMetrics),
+		logger:            logger,
+		ctx:               ctx,
+		cancel:            cancel,
+	}
+
+	// Create a gauge for each possible status (that is not terminal)
+	for _, status := range []commonv2.BlobStatus{
+		commonv2.Queued,
+		commonv2.Encoded,
+		commonv2.GatheringSignatures,
+	} {
+		gauge := prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Name: "dynamodb_blob_metadata_v2_status_count",
+				Help: "Current number of blobs in this status. In case of timeouts or failures when querying DynamoDB (e.g. when there are too many blobs), the last known value will be returned as stale data.",
+				ConstLabels: prometheus.Labels{
+					"status": status.String(),
+				},
+			},
+		)
+		prometheus.MustRegister(gauge)
+		collector.statusMetrics[status] = &BlobStatusMetrics{
+			gauge:        gauge,
+			currentValue: 0,
+		}
+	}
+
+	// Do initial count
+	collector.updateCounts(context.Background())
+
+	return collector
+}
+
+// countBlobsWithStatus counts blobs for a specific status with pagination and timeout handling
+func (collector *DynamoDBCollectorV2) countBlobsWithStatus(ctx context.Context, status commonv2.BlobStatus) (int32, error) {
+	var totalCount int32
+	var cursor *blobstorev2.StatusIndexCursor
+
+	for {
+		select {
+		case <-ctx.Done():
+			return totalCount, ctx.Err()
+		default:
+			blobs, nextCursor, err := collector.blobMetadataStore.GetBlobMetadataByStatusPaginated(ctx, status, cursor, 100)
+			if err != nil {
+				return totalCount, err
+			}
+
+			count := int32(len(blobs))
+			totalCount += count
+
+			collector.logger.Debug("Got partial count for status",
+				"status", status.String(),
+				"partial_count", count,
+				"running_total", totalCount,
+				"has_more", nextCursor != nil,
+			)
+
+			if count == 0 || nextCursor == nil {
+				return totalCount, nil
+			}
+			cursor = nextCursor
+		}
+	}
+}
+
+func (collector *DynamoDBCollectorV2) updateCounts(ctx context.Context) {
+	collector.logger.Debug("Starting blob status count update")
+	startTime := time.Now()
+
+	for status, metrics := range collector.statusMetrics {
+		statusCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
+
+		totalCount, err := collector.countBlobsWithStatus(statusCtx, status)
+		defer cancel()
+
+		if err != nil {
+			if err == context.DeadlineExceeded {
+				collector.logger.Warn("Timeout while counting blobs",
+					"status", status.String(),
+					"current_count", metrics.currentValue,
+					"using_stale_data", true,
+				)
+				continue // Keep using the last known value
+			}
+			collector.logger.Error("Failed to get count of blob metadata by status",
+				"status", status,
+				"err", err,
+				"current_value", metrics.currentValue,
```

Comment:
current_value ->  current_count (or the opposite) to stay consistent

actually it looks there isn't anything special in DeadlineExceeded?